src/MgvLib.sol

```diff
@@ -75,43 +75,42 @@ contract HasMgvEvents {
   event SetGasprice(uint value);
 
   /* Market order execution */
-  event OrderStart();
-  event OrderComplete(
+  // FIXME: This provides the basis information for all following events
+  // FIXME: Observation: All other events follow from this one + the known state of the book
+  event OrderStart(
     address indexed outbound_tkn,
     address indexed inbound_tkn,
     address indexed taker,
-    uint takerGot,
-    uint takerGave,
-    uint penalty,
-    uint feePaid
+    int maxTick,
+    uint fillVolume,
+    bool fillWants
   );
+  // FIXME: got, gave, bounty, and fee can be derived from previous events. 
+  // FIXME: Well, fees aren't actually explicitly included in any event... Could be included here or in OfferSuccess
+  event OrderComplete(uint fee);
 
   /* * Offer execution */
   event OfferSuccess(
-    address indexed outbound_tkn,
```

Comment:
I would maintain indexed `outbound`/`inbound` (later `indexed OLKey` with dynamic ticks) since otherwise indexers cannot ask a node for the events related to a pair.

Reply:
Would you add that to all events that relate to a pair?